File: Array/module1/twoSum.py
"""
Найти 2 числа сумма которых будет равняться искомому числу 
* массив отсортирован в порядке увеличения
Сложность алгоритма О(n) (так как зависит от входных параметров )
"""

from typing import List


def two_sum(arr: List[int], summ: int) -> List[int]:
    left, right = 0, len(arr) - 1

    while right != left:
        temp = arr[left] + arr[right]
        if temp == summ:
            return [arr[left], arr[right]]
        if temp < summ:
            left += 1
            continue
        right -= 1

    return None


def main():
    arr = [1, 3, 5, 6, 7, 8, 9]
    summ = 4
    print(f"Число {arr} из чисел {two_sum(arr, summ)}")
    summ = 7
    print(f"Число {arr} из чисел {two_sum(arr, summ)}")


if __name__ == "__main__":
    main()

File: Array/module1/twoSum.go
// Найти 2 числа сумма которых будет равняться искомому числу 
// * массив отсортирован в порядке увеличения
// Сложность алгоритма О(n) (так как зависит от входных параметров )

package main

import (
	"fmt"
)


func twoSum(arr [7]int, sum int) []int {
	left := 0
	right := len(arr) - 1

	for left != right{
		temp := arr[left] + arr[right]
		
		if temp == sum{
			return []int{arr[left], arr[right]}
		}
		if temp < sum {
			left ++
			continue
		}
		right --
	}
	return nil 
}


func main() {
	array := [7]int{1, 3, 5, 6, 7, 8, 9}
	sum := 4
	fmt.Printf("Число %v состоиз из %v \n", sum, twoSum(array, sum))
	sum = 7
	fmt.Printf("Число %v состоиз из %v \n", sum, twoSum(array, sum))
}